caffe2/onnx/onnx_exporter.cc

```diff
@@ -768,6 +772,79 @@ ConvertedResult OnnxExporter::CreateChannelShuffleNodes(
   return result;
 }
 
+ConvertedResult OnnxExporter::CreateReduceMeanNodes(
+    const caffe2::OperatorDef& def,
+    const std::unordered_map<std::string, caffe2::TensorShape>& shapes) {
+    CAFFE_ENFORCE_GE(def.input_size(), 1);
+    CAFFE_ENFORCE_LE(def.input_size(), 2);
+    CAFFE_ENFORCE_EQ(def.input_size(), 1, "Input \"lengths\" is not supported.");
+    CAFFE_ENFORCE_GE(def.output_size(), 1);
+    const auto& x = def.input(0);
+    const auto& y = def.output(0);
+    const auto& dims = shapes.at(x).dims();
+
+    ConvertedResult result;
+    auto& nodes = result.first;
+    auto& const_tensors = result.second;
+    std::unordered_map<std::string, const caffe2::Argument*> args;
+    for (const auto& a : def.arg()) {
+        args.emplace(a.name(), &a);
+    }
+
+    std::vector<int64_t> axes;
+    int64_t keepdims = 1;
+
+    if (def.type() == "ReduceMean") {
+        // axes
+        {
+            const auto it = args.find("axes");
```

Comment:
using additional block to avoid dup names seems uncommon in PyTorch codebase. Could you remove this trick?

Reply:
This provides more benefit than what you thought.
1. Use `it_suffix` is unnecessary and it makes code becomes longer.
2. With `const` we get SSA, which is easier to read for "local" var.
3. With `{}` I can get even more, i.e. terminate the life span earlier before reassignment. This narrows the "local" semantics and helps people navigating in the code.

Reply:
I didn't see the benefits here...

Especially, the logic here is so simple, and you can just reuse `it` name.

Reply:
Fine...Don't wanna drag this PR longer.
IMHO reuse is bad, and makes the first occurrence asymmetric to the rest.